utilities/load-test/loadtest/loadtest_test.go

```diff
@@ -413,3 +431,102 @@ func TestAllApplication(t *testing.T) {
 	}
 	t.Log("Prometheus, along with 20 guestbook applications are created, deployed and deleted successfully!")
 }
+
+func TestNResource(t *testing.T) {
+	// TEST 1: Deply 100 guestbook in different namespace
+	t.Log("Testing for 100 applications that uses lightweight non-Pod-based resources\n")
+
+	t.Log("\n\nPods Memory Resource before creation:\n\n")
+	memoryInit := GetPodInfo(kubeconfig, masterUrl, namespace, podName)
+	PodInfoParse(memoryInit)
+
+	// Get Pod Restart Count
+	PodC := PodRestartCount(namespace)
+
+	for i := 1; i < 101; i++ {
+		applicationTest := &appv1.Application{
+			TypeMeta: v1.TypeMeta{
+				Kind:       "Application",
+				APIVersion: "argoproj.io/v1alpha1",
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "test-app-" + fmt.Sprintf("%d", i),
+				Namespace: "argocd",
+			},
+			Spec: appv1.ApplicationSpec{
+				Source: appv1.ApplicationSource{
+					Path:           "sample-app",
+					TargetRevision: "HEAD",
+					RepoURL:        "https://github.com/samyak-jn/gitops-service-sample-k8s-resources.git",
+				},
+				Destination: appv1.ApplicationDestination{
+					Namespace: "test-namespace-" + fmt.Sprintf("%d", i),
+					Server:    "https://kubernetes.default.svc",
+				},
+				Project: "default",
+				SyncPolicy: &appv1.SyncPolicy{
+					Automated:   &appv1.SyncPolicyAutomated{},
+					SyncOptions: []string{"CreateNamespace=true"},
+				},
+			},
+		}
+		_, err := GetE2EFixtureK8sClient().AppClientset.ArgoprojV1alpha1().Applications(namespace).Create(context.TODO(), applicationTest, v1.CreateOptions{})
+		if err != nil {
+			t.Errorf("error, %v", err)
+			return
+		}
+	}
+
+	// wait.Poll will keep checking whether a (app variable).Status.Health.Status condition is met
+	err := wait.Poll(1*time.Second, 2*time.Minute, func() (bool, error) {
+
+		// list all Argo CD applications in the namespace
+		appList, err := GetE2EFixtureK8sClient().AppClientset.ArgoprojV1alpha1().Applications(namespace).List(context.TODO(), v1.ListOptions{})
+		if err != nil {
+			t.Errorf("error, %v", err)
+		}
+		// for each application, check if (app variable).Status.Health.Status
+		for _, app := range appList.Items {
+			// if status is empty, return false
+			if app.Status.Health.Status == "" {
```

Comment:
We should wait for all of the applications in the list to have Health Status of 'Healthy' and Sync Status of 'Synced'

Reply:
yep, added a constraint for this